openedx/core/djangoapps/programs/models.py

```diff
@@ -29,31 +43,66 @@ class ProgramsApiConfig(ConfigurationModel):
         )
     )
 
-    PROGRAMS_API_CACHE_KEY = "programs.api.data"
+    enable_student_dashboard = models.BooleanField(
+        verbose_name=_("Enable Student Dashboard Displays"),
+        default=False
+    )
+    enable_studio_tab = models.BooleanField(
+        verbose_name=_("Enable Studio Authoring Interface"),
+        default=False
+    )
 
     @property
     def internal_api_url(self):
         """
-        Generate a URL based on internal service URL and api version number.
+        Generate a URL based on internal service URL and API version number.
         """
-        return urljoin(self.internal_service_url, "/api/v{}/".format(self.api_version_number))
+        return urljoin(self.internal_service_url, '/api/v{}/'.format(self.api_version_number))
 
     @property
     def public_api_url(self):
         """
-        Generate a URL based on public service URL and api version number.
+        Generate a URL based on public service URL and API version number.
+        """
+        return urljoin(self.public_service_url, '/api/v{}/'.format(self.api_version_number))
+
+    @property
+    def authoring_app_config(self):
+        """
+        Returns a named tuple containing information required for working with the Programs
+        authoring app, a Backbone app hosted by the Programs service.
         """
-        return urljoin(self.public_service_url, "/api/v{}/".format(self.api_version_number))
+        js_url = urljoin(self.public_service_url, self.authoring_app_js_path)
+        css_url = urljoin(self.public_service_url, self.authoring_app_css_path)
+
+        return AuthoringAppConfig(js_url=js_url, css_url=css_url)
+
+    @property
+    def is_cache_enabled(self):
+        """Whether responses from the Programs API will be cached."""
+        return self.cache_ttl > 0
+
+    def get_cache_key(self, user):
+        """Construct the appropriate cache key for the given user."""
+        return self.CACHE_KEY_ROOT + '.staff' if user.is_staff else self.CACHE_KEY_ROOT + '.student'
 
     @property
     def is_student_dashboard_enabled(self):
         """
-        Indicate whether LMS dashboard functionality related to Programs should
+        Indicates whether LMS dashboard functionality related to Programs should
         be enabled or not.
         """
         return self.enabled and self.enable_student_dashboard
 
     @property
-    def is_cache_enabled(self):
-        """Whether responses from the Programs API will be cached."""
-        return self.enabled and self.cache_ttl > 0
+    def is_studio_tab_enabled(self):
+        """
+        Indicates whether Studio functionality related to Programs should
+        be enabled or not.
+        """
+        return (
+            self.enabled and
+            self.enable_studio_tab and
+            bool(self.authoring_app_js_path) and
+            bool(self.authoring_app_css_path)
```

Comment:
shouldn't the first two conditions be enough?  it doesn't seem like it would make sense to enable the studio tab without having valid settings for the paths, though maybe that's difficult to validate?